File: task3/main.py
import psycopg2
import json

class User:
    def __init__(self, user_id, name, age, phone):
        self.user_id = user_id
        self.name = name
        self.age = age
        self.phone = phone

    def to_dict(self):
        return {
            "user_id": self.user_id,
            "name": self.name,
            "age": self.age,
            "phone": self.phone
        }

class Response:
    def __init__(self, status_code, data):
        self.status_code = status_code
        self.data = data

    def to_dict(self):
        return {
            "status_code": self.status_code,
            "data": [user.to_dict() for user in self.data]
        }

def main():
    conn = psycopg2.connect(host="localhost", port=5432, user="postgres", password="postgres", dbname="mydb")
    cur = conn.cursor()

    cur.execute("SELECT user_id, name, age, phone FROM public.user_table")

    users = []
    for row in cur.fetchall():
        user = User(row[0], row[1], row[2], row[3])
        users.append(user)

    response = Response(200, users)

    jsonResponse = json.dumps(response.to_dict())
    print(jsonResponse)

    cur.close()
    conn.close()

if __name__ == "__main__":
    main()
  

File: task3/main.go
package main

import (
    "database/sql"
    "fmt"
    "log"
    "encoding/json"
    _ "github.com/lib/pq"
)

type User struct {
    UserId int `json:"user_id"`
    Name string `json:"name"`
    Age int `json:"age"`
    Phone sql.NullString `json:"phone,omitempty"`
}

type Response struct {
    StatusCode int `json:"status_code"`
    Data []User `json:"data"`
}

func main() {
    db, err := sql.Open("postgres", "host=localhost port=5432 user=postgres password=postgres dbname=mydb sslmode=disable")
    if err != nil {
        log.Fatal(err)
    }
    defer db.Close()

    rows, err := db.Query("SELECT user_id, name, age, phone FROM public.user_table")
    if err != nil {
        log.Fatal(err)
    }
    defer rows.Close()

    var users []User
    for rows.Next() {
        var user User
        err := rows.Scan(&user.UserId, &user.Name, &user.Age, &user.Phone)
        if err != nil {
            log.Fatal(err)
        }
        users = append(users, user)
    }
    if err := rows.Err(); err != nil {
        log.Fatal(err)
    }

    response := Response{
        StatusCode: 200,
        Data: users,
    }

    jsonResponse, err := json.Marshal(response)
    if err != nil {
        log.Fatal(err)
    }

    fmt.Println(string(jsonResponse))
}
